tests/test_decorators.py

```diff
@@ -100,3 +101,46 @@ def add(a, b):
         self.assertEqual(add.cache, {(1, 2): 3, (2, 3): 5})
         self.assertEqual(add(3, 4), 7)
         self.assertEqual(add.cache, {(1, 2): 3, (2, 3): 5, (3, 4): 7})
+
+    def test_word_replace(self):
+        @WordReplace("cloud", "butt")
+        def word_counter(text):
```

Comment:
This is something complicated to explain... You're defining a business-logic intensive function inside the test case. If the test case fails, how can you be sure it's not because you've failed the function? Usually it's better to have really dumb test cases.